src/goose/brokers/broker.clj

```diff
@@ -6,3 +6,35 @@
 
 (defmethod new :default [x & _]
   (throw (ex-info "Invalid broker type" x)))
+
+(defprotocol Broker
+  "Protocol for all Goose brokers."
+  (enqueue [this job])
+  (schedule [this schedule job])
+  (start [this worker-opts])
```

Comment:
Okay, I see that `goose.worker/start` returns an object that implements a separate `Shutdown` protocol.

Rather than this, how about the following implementation:
1. Have two protocols: one called `Broker` / `Client` / `Producer` and one called `Worker` / `Consumer`.
2. The two protocols can have their own separate `start`/`stop` functions and manage their own state. In the case of the `Producer`, it might manage a connection pool. The `Worker` / `Consumer` might manage various thread pools.
3. This allows the state to be maintained in the `Producer` / `Worker` records, rather than being closed over by an anonymous `stop` function. This will allow inspecting the state for easier debugging, and I think it also makes for a more intuitive API.